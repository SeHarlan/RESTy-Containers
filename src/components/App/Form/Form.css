.Form {
  background-color: slategray;
  section {
    margin: 1rem;
  }
  label {
    margin: 1rem ;
  }
}