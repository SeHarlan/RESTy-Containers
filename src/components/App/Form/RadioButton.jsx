import React from 'react';
import PropTypes from 'prop-types';

const RadioButton = ({ value, selectedOption, onOptionChange }) => (
  <label>
    <input checked={selectedOption === value} type="radio" name="method" value={value} onChange={onOptionChange} />
    {value}
  </label>
);
RadioButton.propTypes = {
  selectedOption: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onOptionChange: PropTypes.func.isRequired
};
export default RadioButton;
