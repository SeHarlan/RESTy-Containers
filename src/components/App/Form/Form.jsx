import React from 'react'; 
import PropTypes from 'prop-types';
import RadioButton from './RadioButton';
import style from './Form.css';


const Form = ({ onSubmit, urlInput, jsonInput, selectedOption = 'GET', onOptionChange, onJsonChange, onUrlChange }) => {

  const radios = ['GET', 'POST', 'PUT', 'PATCH', 'DELETE'].map(method => (
    <RadioButton key={method} value={method} selectedOption={selectedOption} onOptionChange={onOptionChange} />));

  return (
    <form className={style.Form} onSubmit={onSubmit}>

      <input type="text" name="url" value={urlInput} onChange={onUrlChange} placeholder="URL"/>
      <section>

        {radios}
        <button type="submit">Fetch</button>
      </section>
    
      <textarea name="jsonBody" value={jsonInput} cols="50" rows="10" onChange={onJsonChange} placeholder="Raw JSON body"></textarea>
    </form>);
};

Form.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  urlInput: PropTypes.string.isRequired,
  jsonInput: PropTypes.string,
  selectedOption: PropTypes.string,
  onOptionChange: PropTypes.func.isRequired,
  onJsonChange: PropTypes.func.isRequired,
  onUrlChange: PropTypes.func.isRequired
};
export default Form;
