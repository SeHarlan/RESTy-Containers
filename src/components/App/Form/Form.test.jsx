import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('renders Form', () => {
    const wrapper = shallow(<Form 
      onSubmit={() => {}}
      urlInput="url"
      jsonInput="url"
      selectedOption="GET"
      onOptionChange={()=>{}}
      onJsonChange={()=>{}}
      onUrlChange={()=>{}} 
      onBearerTokenChange={()=>{}}
      onNameChange={()=>{}}
      onPasswordChange={()=>{}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
