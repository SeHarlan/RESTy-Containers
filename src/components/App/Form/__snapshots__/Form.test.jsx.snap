// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<form
  className="Form"
  onSubmit={[Function]}
>
  <input
    name="url"
    onChange={[Function]}
    placeholder="URL"
    type="text"
    value="url"
  />
  <section>
    <RadioButton
      key="GET"
      onOptionChange={[Function]}
      selectedOption="GET"
      value="GET"
    />
    <RadioButton
      key="POST"
      onOptionChange={[Function]}
      selectedOption="GET"
      value="POST"
    />
    <RadioButton
      key="PUT"
      onOptionChange={[Function]}
      selectedOption="GET"
      value="PUT"
    />
    <RadioButton
      key="PATCH"
      onOptionChange={[Function]}
      selectedOption="GET"
      value="PATCH"
    />
    <RadioButton
      key="DELETE"
      onOptionChange={[Function]}
      selectedOption="GET"
      value="DELETE"
    />
    <button
      type="submit"
    >
      Fetch
    </button>
  </section>
  <section
    className="Json"
  >
    <textarea
      cols="30"
      name="json"
      onChange={[Function]}
      placeholder="Raw JSON body"
      rows="10"
      value="url"
    />
    <div
      className="Headers"
    >
      <h3>
        Basic Authorization
      </h3>
      <span>
        Username: 
      </span>
      <input
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="text"
      />
      <span>
        Password: 
      </span>
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="text"
      />
      <span>
        Bearer Token: 
      </span>
      <input
        name="bearerToken"
        onChange={[Function]}
        placeholder="Bearer Token"
        type="text"
      />
    </div>
  </section>
</form>
`;
